flexmeasures/auth/policy.py

```diff
@@ -226,13 +227,21 @@ def can_modify_role(user, roles_to_modify) -> bool:
             return False
         if role.name == ADMIN_READER_ROLE and not user.has_role(ADMIN_ROLE):
             return False
-        if role.name == ACCOUNT_ADMIN_ROLE and not (
-            user.has_role(ADMIN_ROLE) or user.has_role(CONSULTANT_ROLE)
-        ):
-            return False
-        if role.name == CONSULTANT_ROLE and not (
-            user.has_role(ADMIN_ROLE) or user.has_role(ACCOUNT_ADMIN_ROLE)
-        ):
-            return False
+        if role.name == ACCOUNT_ADMIN_ROLE:
+            if user.has_role(ADMIN_ROLE):
+                return True
```

Comment:
No, never return `True` in this loop while you are testing a list of roles. Only return `False`, the once `True` at the end.

I'd keep this to one if statement, probably.

Or refactor this function to test only one role at a time - maybe that makes for clearer code. We could return `True` only for the conditions we actually support, and return `False` at the end if none of the conditions were met.  (role==ADMIN and role==ADMIN_READER could be handled at the beginning just as now, of course)'